components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/util/ApiKeyAuthConfig.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.conditional.auth.functions.http.util;
+
+import org.apache.http.client.methods.HttpUriRequest;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+
+import java.util.Map;
+
+import static org.wso2.carbon.identity.conditional.auth.functions.common.utils.CommonUtils.*;
+
+/**
+ * Implementation of the {@link AuthConfig}
+ * This class is used to configure the API key authentication.
+ * The API key is added to the request header.
+ */
+public class ApiKeyAuthConfig implements AuthConfig {
+    private String headerName = "X-API-KEY"; // Default header name
+    private String apiKey;
+    private static final String HEADER_NAME_VARIABLE_NAME = "headerName";
+    private static final String API_KEY_VARIABLE_NAME = "apiKey";
+
+    public void setHeaderName(String headerName) throws SecretManagementException {
+        this.headerName = getResolvedSecret(headerName);
```

Comment:
header is not a secret. In a defined function I don't think we need to have this.

Or else secret handling should be sorted centrally before this function gets invoked

Reply:
If it's added as a secret, it will be resolved as secret or else it'll be return the same value.

Reply:
Let's remove secret handling logic for the time being

Reply:
Fixed with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/157/commits/05c4f77e8d18ba911c3b98bc7e18ec3fb73dcd33